File: SDKs/github.com/ghes-2.21/1.1.4/go/pkg/models/operations/gistsgetrevision.go
package operations

import (
"openapi/pkg/models/shared")

type GistsGetRevisionPathParams struct {
    GistID string `pathParam:"style=simple,explode=false,name=gist_id"`
    Sha string `pathParam:"style=simple,explode=false,name=sha"`
    
}

type GistsGetRevisionRequest struct {
    PathParams GistsGetRevisionPathParams 
    
}

type GistsGetRevisionResponse struct {
    ContentType string 
    StatusCode int64 
    BasicError *shared.BasicError 
    GistSimple *shared.GistSimple 
    ValidationError *shared.ValidationError 
    
}

File: SDKs/github.com/api.github.com/1.1.4/python/src/sdk/models/operations/gists_get_revision.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class GistsGetRevisionPathParams:
    gist_id: str = field(metadata={'path_param': { 'field_name': 'gist_id', 'style': 'simple', 'explode': False }})
    sha: str = field(metadata={'path_param': { 'field_name': 'sha', 'style': 'simple', 'explode': False }})
    

@dataclass
class GistsGetRevisionRequest:
    path_params: GistsGetRevisionPathParams = field()
    

@dataclass
class GistsGetRevisionResponse:
    content_type: str = field()
    status_code: int = field()
    basic_error: Optional[shared.BasicError] = field(default=None)
    gist_simple: Optional[shared.GistSimple] = field(default=None)
    validation_error: Optional[shared.ValidationError] = field(default=None)
    
